api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,486 @@
+        - name: batch_keep_alive_limit
+          in: query
+          description: |
+            Maximum number of keep-alive messages to get in a row
+            before closing the connection.
+            Unlimited by default.
+          type: integer
+          format: int32
+          required: false
+        - name: Accept-Encoding
+          in: header
+          required: false
+          description: '"If set the client requests the body to be gzip compressed. For Example "Content-Encoding: gzip"'
+          type: string
+      responses:
+        '200':
+          description: |
+            Starts streaming to the client.  
+            Stream format is a continuous series of
+            `SimpleStreamEvent`s separated by `\n`
+          schema:
+            $ref: '#/definitions/SimpleStreamEvent'
+        '400':
+          description: Bad syntax
+          schema:
+            $ref: '#/definitions/Problem'
+        '401':
+          description: Not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '404':
+          description: Not found. Typically in case of requesting from a non existing topic of partition.
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Not found. Typically in case of requesting from a non existing topic of partition.
+          schema:
+            $ref: '#/definitions/Problem'
+
+
+# ################################### #
+#                                     #
+#             Definitions             #
+#                                     #
+# ################################### #
+
+definitions:
+  Event:
+    type: object
+    description: |
+      This is the most general representation of an event, that can be processed
+      by Nakadi.
+
+      It should be used as a base definition for all events, that flow through
+      Nakadi by extending attributes of this object type.
+    properties:
+      event_type:
+        type: string
+        example: 'https://resource-events.zalando.com/ResourceCreated'
+      partitioning_key:
+        type: string
+        example: 'ABC123XXX-001'
+      metadata:
+        $ref: '#/definitions/EventMetaData'
+  EventMetaData:
+    type: object
+    properties:
+      id:
+        type: string
+        format: uuid
+      created:
+        type: string
+        format: data-time
+      root_id:
+        type: string
+        format: uuid
+      parent_id:
+        type: string
+        format: uuid
+      scopes:
+        type: array
+        items:
+          type: string
+  Problem:
+    type: object
+    required:
+      - detail
+    properties:
+      detail:
+        type: string
+        description: Problem description
+        example: Topic does not exist
+  Metrics:
+    type: object
+  Topic:
+    type: object
+    required:
+      - name
+    properties:
+      name:
+        type: string
+        description: Topic name
+        example: article.120
```

Comment:
IMO topics are owned by applications that also define the application specific event scheme -- see comment above.
So owner_application_id could be an interesting additional topic attribute.
Also number of partitions is an interesting topic property that should not be hided.